lookout/style/format/benchmarks/quality_report_noisy.py

```diff
@@ -344,17 +343,26 @@ def quality_report_noisy(bblfsh: str, language: str, confidence_threshold: float
             n_rules_filtered[repo] = len(rules_id)
             # compute the confidence and recall limit for the given precision threshold
             for i, (prec, rec) in enumerate(zip(precisions[repo], recalls[repo])):
-                if prec < precision_threshold:
-                    break
-                confidence_threshold_exp[repo] = (i, round(rules_id[i][1], 4))
-                rec_threshold_prec[repo] = rec
+                if prec >= precision_threshold:
+                    accepted_rules[repo].append((i, rules_id[i][1], rec))
+            last_accepted_rule[repo] = min(accepted_rules[repo], key = lambda r: r[1])
+            confidence_threshold_exp[repo] = (last_accepted_rule[repo][0], last_accepted_rule[repo][1])
     finally:
         client._channel.close()
 
+    # compute the index of the last accepted rule according to the maximum confidence threshold
+    limit_conf_id = {}
+    max_confidence_threshold_exp = max(confidence_threshold_exp.values(), key = lambda x: x[1])
```

Comment:
`from operations import itemgetter`

```suggestion
    max_confidence_threshold_exp = max(confidence_threshold_exp.values(), key=itemgetter(1))
```